### ISSUE TITLE
keyhunt - Multiple Processes

### ISSUE THREAD

**Issue by luckyalab** [albertobsd/keyhunt#41]
Hi,

Instead of running one process like

sudo ./keyhunt -m bsgs -f  tests/120.txt   -t 4 -k 1024 -q -R

Can we run 4 parallel process like this?
sudo ./keyhunt -m bsgs -f  tests/120A.txt   -t 1 -k 256 -q -R
sudo ./keyhunt -m bsgs -f  tests/120B.txt   -t 1 -k 256 -q -R
sudo ./keyhunt -m bsgs -f  tests/120C.txt   -t 1 -k 256 -q -R
sudo ./keyhunt -m bsgs -f  tests/120D.txt   -t 1 -k 256 -q -R

Because  processing 1761607680 bP points for  -k 1024  takes a very long time.

Please advise

Thank you
Regards

--

**Comment by albertobsd** [comment]
Yes you can run many instances of the process as you like. But just think that RAM is not shared bewteen process. Also each process is goint to manage his own bPTable

**Comment by luckyalab** [comment]
> 
> 
> Yes you can run many instances of the process as you like. But just think that RAM is not shared bewteen process. Also each process is goint to manage his own bPTable

Thank you @albertobsd  

As followup to your comment  "RAM is not shared between process" 
What is the performance impact of using SWAP on SSD disk? 

TIA
Regards

**Comment by albertobsd** [comment]
I don't know because i never tested it, im planning to write the bloom filter in file and checkit from file without load it in RAM, but the disk need to be a NVMe Disk to get a decent performance. But im still planning it, im not developed anything relate with that

**Comment by luckyalab** [comment]
> 
> 
> I don't know because i never tested it, im planning to write the bloom filter in file and checkit from file without load it in RAM, but the disk need to be a NVMe Disk to get a decent performance. But im still planning it, im not developed anything relate with that

That will be great if you can write the bloom filter to a file. NVMe disk are available :)

**Comment by luckyalab** [comment]
Hi @albertobsd 

I was trying to run these 2 parallel process.
First I started one process
sudo ./keyhunt -m bsgs -f  tests/120.txt   -t 1 -k 80 -q -R
After It started processing I started the second process
sudo ./keyhunt -m bsgs -f  tests/120.txt   -t 1 -k 80 -q -R

What  I observed is that the in the first process the Total is not changing even though keys per second is changing
![image](https://user-images.githubusercontent.com/82208222/114745547-48cf2400-9d3e-11eb-9aff-a5247f98578b.png)

The second process screenshot at the same time
![image](https://user-images.githubusercontent.com/82208222/114745728-7ddb7680-9d3e-11eb-8802-590df2e53dd7.png)

Thank you
Regards


--

**Comment by luckyalab** [comment]
HI  @albertobsd 
Update on the parallel process.

when I run 2 process simultaneously  the Total keys does not change for both process.

please find screenshots for your reference.

Process 1screenshot
![image](https://user-images.githubusercontent.com/82208222/114756042-d8c69b00-9d49-11eb-8885-f1ccc07785fb.png)

Process 2 screen shot

![image](https://user-images.githubusercontent.com/82208222/114756379-33f88d80-9d4a-11eb-9f04-a7bf2fb4abb8.png)

thank you and please advise
regards

**Comment by albertobsd** [comment]
What is your CPU?

**Comment by luckyalab** [comment]
> 
> 
> What is your CPU?
Intel(R) Core(TM) i7-4600U CPU @ 2.10GHz   2.69 GHz
2 cores and 4 logical processors 8 GB RAM

Thank you
Regards
